src/main/java/com/adyen/service/TerminalCloudAPI.java

```diff
@@ -83,4 +82,18 @@ public TerminalAPIResponse sync(TerminalAPIRequest terminalAPIRequest) throws IO
         }.getType());
         return terminalAPIResponse;
     }
+
+    /**
+     * Checks if SaleToAcquirerData is set. If it is not creates an encoded base 64 string with application info.
+     *
+     * @param terminalAPIRequest TerminalAPIRequest
+     * @return TerminalAPIRequest
+     */
+    private static TerminalAPIRequest setSaleToAcquirerDataModel(TerminalAPIRequest terminalAPIRequest) {
+        if (terminalAPIRequest != null && !terminalAPIRequest.isSaleToAcquirerDataModelSet()) {
+            SaleToAcquirerDataModel saleToAcquirerDataModel = new SaleToAcquirerDataModel();
+            terminalAPIRequest.getSaleToPOIRequest().getPaymentRequest().getSaleData().setSaleToAcquirerData(saleToAcquirerDataModel.toBase64());
+        }
+        return terminalAPIRequest;
```

Comment:
what if is is manually set by the merchant ? do we then not overwrite it or add it with our library info ?